routes/volunteers/volunteers.js

```diff
@@ -1,15 +1,55 @@
 const express = require('express');
+const Fuse = require('fuse.js');
 const { pool } = require('../../server/db');
-const { isNumeric, keysToCamel, getUsersQuery } = require('../utils');
+const { isNumeric, keysToCamel } = require('../utils');
 
 const volunteerRouter = express();
 
 // get all volunteers
 volunteerRouter.get('/', async (req, res) => {
   try {
-    const conditions = `WHERE users.role = 'volunteer'`;
-    const volunteers = await pool.query(getUsersQuery(conditions));
-    res.status(200).json(keysToCamel(volunteers.rows));
+    const { driverOption, ageOption, eventInterest, searchQuery } = req.query;
+    let driverCondition = 'TRUE';
+    let ageCondition = 'TRUE';
+    let eventCondition = 'TRUE';
+
+    if (driverOption === 'Can Drive') {
+      driverCondition = ' users.can_drive';
+    } else if (driverOption === 'Cannot Drive') {
+      driverCondition = ' NOT users.can_drive';
+    }
+
+    if (ageOption === 'Adult') {
+      ageCondition = `date_part('year', age(users.birthdate)) >= 18`;
+    } else if (ageOption === 'Minor') {
+      ageCondition = `date_part('year', age(users.birthdate)) < 18`;
+    }
+
+    if (eventInterest === 'Distributions') {
+      eventCondition = ' users.distribution_interest';
+    } else if (driverOption === 'Food Running') {
+      eventCondition = ' users.food_runs_interest';
+    }
```

Comment:
can we convert these `let` variables and the following `if/else` loops to mappings/dictionaries pls - also we need a default case if the `driverOption` or `eventInterest` isn't one of the two values (which you can probably use the `TRUE` strings for)